src/inmanta/validation_type.py

```diff
@@ -187,7 +187,7 @@ def validate_type(
 def _cachable_validate_type(fq_type_name: str, validation_parameters: HashKeyContainer) -> pydantic.TypeAdapter[object]:
     module_name, type_name = fq_type_name.split(".", 1)
     module = importlib.import_module(module_name)
-    requested_type: object = getattr(module, type_name)
+    requested_type: type = getattr(module, type_name)
```

Comment:
It's not necessarily a type though. It may also be a callable. And well, it could be anything really. Perhaps it's rather the `parametrize_type` signature that is too narrow here.

Reply:
I left this as it was